components/places/src/api/places_api.rs

```diff
@@ -222,14 +222,61 @@ impl PlacesApi {
         }
     }
 
-    // TODO: We need a better result here so we can return telemetry.
-    // We possibly want more than just a `SyncTelemetryPing` so we can
-    // return additional "custom" telemetry if the app wants it.
+    // NOTE: These should be deprecated as soon as possible - that will be once
+    // all consumers have been updated to use the .sync() method below, and/or
+    // we have implemented the sync manager and migrated consumers to that.
     pub fn sync_history(
         &self,
         client_init: &sync15::Sync15StorageClientInit,
         key_bundle: &sync15::KeyBundle,
     ) -> Result<telemetry::SyncTelemetryPing> {
+        self.do_sync_one(
+            "history",
+            move |conn, mem_cached_state, disk_cached_state| {
+                let interruptee = conn.begin_interrupt_scope();
+                let store = HistoryStore::new(&conn, &interruptee);
+                sync_multiple(
+                    &[&store],
+                    disk_cached_state,
+                    mem_cached_state,
+                    client_init,
+                    key_bundle,
+                    &interruptee,
+                )
+            },
+        )
+    }
+
+    pub fn sync_bookmarks(
+        &self,
+        client_init: &sync15::Sync15StorageClientInit,
+        key_bundle: &sync15::KeyBundle,
+    ) -> Result<telemetry::SyncTelemetryPing> {
+        self.do_sync_one(
+            "bookmarks",
+            move |conn, mem_cached_state, disk_cached_state| {
+                let interruptee = conn.begin_interrupt_scope();
+                let store = BookmarksStore::new(&conn, &interruptee);
+                sync_multiple(
+                    &[&store],
+                    disk_cached_state,
+                    mem_cached_state,
+                    client_init,
+                    key_bundle,
+                    &interruptee,
+                )
+            },
+        )
+    }
+
+    pub fn do_sync_one<F>(
+        &self,
+        name: &'static str,
+        syncer: F,
+    ) -> Result<telemetry::SyncTelemetryPing>
```

Comment:
I think this `Result` is OK to leave, though. #1112 removes it, but I can take care of it when rebasing atop yours.

Reply:
These return values are really just for b/w compatibility - I'd like to see this die asap.